Source/Core/Credit.js

```diff
@@ -71,38 +52,74 @@ define([
-        var showOnScreen = defaultValue(options.showOnScreen, false);
+            var hasLink = (defined(link));
+            var hasImage = (defined(imageUrl));
+            var hasText = (defined(text));
 
-        var hasLink = (defined(link));
-        var hasImage = (defined(imageUrl));
-        var hasText = (defined(text));
+            //>>includeStart('debug', pragmas.debug);
+            if (!hasText && !hasImage && !hasLink) {
+                throw new DeveloperError('options.text, options.imageUrl, or options.link is required.');
+            }
+            //>>includeEnd('debug');
 
-        //>>includeStart('debug', pragmas.debug);
-        if (!hasText && !hasImage && !hasLink) {
-            throw new DeveloperError('options.text, options.imageUrl, or options.link is required.');
-        }
-        //>>includeEnd('debug');
+            if (!hasText && !hasImage) {
+                text = link;
+            }
 
-        if (!hasText && !hasImage) {
-            text = link;
-        }
+            this._text = text;
+            this._imageUrl = imageUrl;
+            this._link = link;
+            this._hasLink = hasLink;
+            this._hasImage = hasImage;
 
-        this._text = text;
-        this._imageUrl = imageUrl;
-        this._link = link;
-        this._hasLink = hasLink;
-        this._hasImage = hasImage;
-        this._showOnScreen = showOnScreen;
+            var element = document.createElement('span');
+            var a;
+            if (hasImage) {
+                var content = document.createElement('img');
+                content.src = imageUrl;
+                if (defined(text)) {
+                    content.alt = text;
+                    content.title = text;
+                }
+                if (hasLink) {
+                    a = document.createElement('a');
+                    a.appendChild(content);
+                    a.href = link;
+                    a.target = '_blank';
+                    element.appendChild(a);
+                } else {
+                    element.appendChild(content);
+                }
+                element.className = 'cesium-credit-image';
+            } else {
+                if (hasLink) {
+                    a = document.createElement('a');
+                    a.textContent = text;
+                    a.href = link;
+                    a.target = '_blank';
+                    element.appendChild(a);
+                } else {
+                    element.textContent = text;
+                }
+                element.className = 'cesium-credit-text';
+            }
 
-        // Credits are immutable so generate an id to use to optimize equal()
-        var id;
-        var key = JSON.stringify([text, imageUrl, link]);
+            html = '<span>' + element.innerHTML + '</span>';
+            key = JSON.stringify([text, imageUrl, link]);
+        } else {
+            key = html;
+        }
 
         if (defined(creditToId[key])) {
```

Comment:
Do we even need `creditToId` anymore?  Can't we just use the `html` as the id directly?

Reply:
`credit.id` is a number and is used as the index of the credit in the array of credits in `CreditDisplay`.  So we still need this.